lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -3621,4 +3641,13 @@ public function CustomFunctionsReturningStrings()
 
         return $function;
     }
+
+    private function getMetadataForDqlAlias(string $dqlAlias): ClassMetadata
+    {
+        if (! isset($this->queryComponents[$dqlAlias]['metadata'])) {
+            throw new LogicException(sprintf('No metadata for DQL alias: %s', $dqlAlias));
```

Comment:
Even though this is a LogicException, it can cause BC problems. Why introducing it in 2.x?

Reply:
The pieces of code that I've replaced with a call to this method did previously access the `metadata` without prior check and afterwards assume that the result is a `ClassMetadata` instance.

**if** that `LogicException` is ever raised (I don't expect it to), it is raised on execution paths that triggered hard errors previously anyway. This is why I don't expect this change to be a BC break.